mcstatus/motd.py

```diff
@@ -0,0 +1,549 @@
+            except KeyError:
+                if item["color"] == "reset":
+                    # minecraft server actually can't return {"reset": True}, instead
+                    # it will return {"color": "reset"}. but for follow naming, we
+                    # will not move `Formatting.RESET` to `MinecraftColor`.
+                    #
+                    # see https://wiki.vg/Chat#Shared_between_all_components, `color` field
+                    parsed_motd.append(Formatting.RESET)
+                else:
+                    parsed_motd.append(WebColor.parse(hex=item["color"]))
+
+        for style_key, style_val in Formatting.__members__.items():
+            if item.get(style_key.lower()) is False or item.get(style_key.lower()) == "false":
+                parsed_motd.remove(style_val)
+            elif item.get(style_key.lower()) is not None:
+                parsed_motd.append(style_val)
+
+        text = item.get("text")
+        if text is not None:
+            parsed_motd.append(text)
+        translate = item.get("translate")
+        if text is None and translate is not None:
+            parsed_motd.append(TranslateString(translate))
+        parsed_motd.append(Formatting.RESET)
+
+        if item.get("extra"):
+            auto_add = list(filter(lambda e: type(e) is Formatting and e != Formatting.RESET, parsed_motd))
+
+            for element in item["extra"]:
+                parsed_motd.extend(cls._parse_as_dict(element, auto_add=auto_add.copy()))
+
+        return parsed_motd
+
+    @staticmethod
+    def _parse_as_str(raw: str, *, bedrock: bool = False) -> list[ParsedMotdItem]:
+        """Parse a MOTD when it's string.
+
+        .. note:: This method returns a lot of empty strings, use ``Motd.simplify`` to remove them.
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition?
+            Ignores ``MinecraftColor.MINECOIN_GOLD`` if it's ``False``.
+        :returns: ``ParsedMotdItem`` list, which need to be passed to ``__init__``.
+        """
+        parsed_motd: list[ParsedMotdItem] = []
+
+        split_raw = MOTD_COLORS_RE.split(raw)
+        for element in split_raw:
+            clean_element = element.lstrip("&§")
+            if clean_element == "g" and not bedrock:
+                parsed_motd.append(element)  # minecoin_gold on java server
+                continue
+            try:
+                parsed_motd.append(MinecraftColor(clean_element))
+            except ValueError:
+                try:
+                    parsed_motd.append(Formatting(clean_element))
+                except ValueError:
+                    # just a text
+                    parsed_motd.append(element)
+
+        return parsed_motd
+
+    def simplify(self) -> Self:
+        """Simplify MOTD by removing unused formatting, unaffected colors etc.
+
+        :returns: ``self`` parameter, so you can chain this method.
```

Comment:
I don't think we need to support chaining here, it's usually more pythonic to just return None for methods which modify the instance, returning the instance might be misleading as people could assume that this creates a new motd, rather than mutating the existing one.

Reply:
I thought, that this will be used as
```py
return motd.simplify().minecraft
```
I actually quite hate returning `None` in methods, that better to use in chain. This still confuses me in builtins literals.

Reply:
My point stands with the above example, as to me it immediately seems like `simplify` just created a new instance here which we called minceraft for, so I'd assume the original `motd` was unmodified and is still in the unsmplified motd. Chaining should only be reserved for some special cases where it would be very annoying not to have it, because there's a lot of functions to be called that chain on each other, but for a single call case this is not necessary and needlessly confusing.

Reply:
[I changed the behavior](https://github.com/py-mine/mcstatus/compare/ffe1cc869956fc92fc7980d4c8a265d23adfa1b6..ae2dd4eff4d605ae236fba6a2f87e783aa51de2e#diff-c462c16c38f067b712f30e9e5d65eda0a525e97348b12492eacf676da5656ed6L310-R329) in this function, to create a new instance instead of modifying the exist one.